src/frontend/contexts/ActiveProjectIdStoreContext.tsx

```diff
@@ -0,0 +1,99 @@
+import {createContext, useContext} from 'react';
+import {createStore, useStore, type StoreApi} from 'zustand';
+import {
+  createJSONStorage,
+  persist as createPersistedState,
+} from 'zustand/middleware';
+
+import {MMKVZustandStorage} from '../hooks/persistedState/createPersistedState';
+
+type ActiveProjectIdState = {
+  projectId?: string;
+};
+
+// NOTE: Do not change!
+const STORAGE_KEY = 'ActiveProjectId' as const;
+
+// Zustand's persist middleware and using `createJSONStorage()` assumes that states are represented as objects.
+// Using a scalar value requires tedious workarounds that are more trouble than shaping the state according to Zustand's assumptions.
+// https://github.com/pmndrs/zustand/blob/17e281fd75a8200e3598658e732b8b4a3055f0b1/src/middleware/persist.ts#L181-L184
+function createInitialState(): ActiveProjectIdState {
+  return {};
+}
+
+export function createActiveProjectIdStore({persist} = {persist: false}) {
+  let store: StoreApi<ActiveProjectIdState>;
+
+  if (persist) {
+    store = createStore(
+      createPersistedState(createInitialState, {
+        name: STORAGE_KEY,
+        storage: createJSONStorage(() => MMKVZustandStorage),
+        version: 1,
+        migrate: (persistedState, version): ActiveProjectIdState => {
+          const newState = createInitialState();
+
+          if (version === 0) {
+            if (typeof persistedState !== 'object') return newState;
+            if (persistedState === null) return newState;
+
+            if (
+              'projectId' in persistedState &&
+              typeof persistedState.projectId === 'string'
+            ) {
+              return {
+                projectId: persistedState.projectId,
+              };
+            }
+          }
+
+          return newState;
+        },
+      }),
+    );
```

Comment:
overall it looks good, but before i submit an official review I just want to clarify what this migration is for? Is it because previously the "action" was in the store?

Reply:
it's to make sure that the previously persisted active project id is actually ported over to the new storage version since we're bumping the version. without doing this, zustand will basically act as though no previous value was persisted and fall back to the initial value we set.

as noted in my PR description, not sure how imperative doing this is in terms of user impact. if we don't do it, we just have to rely on the logic that's done [here](https://github.com/digidem/comapeo-mobile/blob/1861dab49b42398a3143d5cd9dc67373a38f8e4f/src/frontend/contexts/ActiveProjectContext.tsx#L35-L62), which works okay for now but is a bit fragile

Reply:
alternatively, we can use the same storage but not bump the version. I think the shape of what's stored doesn't change with this PR, as I believe functions within an object (like the original `setProjectId()`) don't get persisted due to serialization via JSON stringify.

It's a bit riskier but might be okay. The main question is do we want to start with a clean slate or try to use the existing storage

Reply:
Yeah, sorry, what I was trying to get at is understanding why we need to bump the version. 

Even if the shape of the store changes (but I think your right, setProjectId doesn't get persisted), both stores would still have the value `projectId`. so I *think* it would be fine. BUT even if it isn't fine we do have that `useEffect` check.

The complexity of this migration feels a little unnecessary and it ill become less of a problem when we have multi projects down the line

Reply:
Yeah makes sense. can update to omit the migration and test that it works!

Reply:
Addressed via https://github.com/digidem/comapeo-mobile/pull/1061/commits/37b12e7a4e6c36d5c04847d647a79c9373e65e10

Just to clarify: this keeps the same persistence key but no longer bumps the storage version. is that sufficient, or should we just update the persistence key and start fresh?

Reply:
> Just to clarify: this keeps the same persistence key but no longer bumps the storage version. is that sufficient, or should we just update the persistence key and start fresh?

I think keeping the same persistence key is the easiest thing to do, and i don't think it will cause any problems